### ISSUE TITLE
Idea for token generation - docker entrypoint

### ISSUE THREAD

**Issue by cameronbraid** [travisghansen/external-auth-server#72]
Currently token generation requires editing source files. 

One idea I may suggest is to add an additional entrypoint into the docker container for token generation and you write the config to stdin and read the generated token from stdout

I'm looking to use this project in a kubernetes cluster with istio, and I think writing a kustomize generator would be a good integration point for token generation - so calling into the docker container from the kustomize plugin seems like a good option

**Comment by travisghansen** [comment]
@cameronbraid thanks for the interest! I hope it works well!

I'm fully open to this kind of thing for sure. Should be easy enough.

I honestly never intended the project to be used directly for token generation...but I've never documented other tools/languages that could be used either. `jwt` is pretty prolific and the encryption scheme is standard as well (although it's not using `jwt` encryption).

In the long-term as time permits I think it would be interesting to build a 'token server' and have a UI to fully manage the tokens with.

**Comment by cameronbraid** [comment]
Done :)

https://github.com/travisghansen/external-auth-server/pull/74

**Comment by cameronbraid** [comment]
EAS is just what I have been looking for.

Was easy to get going in the cluster and easy to configure and customize.

I've been able to implement a kustomize generator that transforms the following :

```
apiVersion: kustomize.drivenow.com.au/v1
kind: ExternalAuthServiceEnvoyFilter
metadata:
  name: tekton-dashboard-auth
spec:
  configTokenSignSecret: xxx
  configTokenEncryptSecret: yyy
  token:
    eas:
      plugins:
        - type: htpasswd
          realm: Demo
          users: 
            - username: dev
              password: dev
            - username: cameron
              password: cameron
  sidecar:
    selector: 
      app: tekton-dashboard
    port: 9097
```

into the envoy filter yaml as per

```
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: tekton-dashboard-auth
  namespace: tekton-pipelines
spec:
  filters:
    - filterConfig:
        failure_mode_allow: false
        httpService:
          authorizationRequest:
            allowedHeaders:
              patterns:
                - exact: cookie
                - exact: X-Forwarded-Host
                - exact: X-Forwarded-Method
                - exact: X-Forwarded-Proto
                - exact: X-Forwarded-Uri
            headers_to_add:
              - key: x-eas-verify-params
                value: >-
                  {"config_token":"...."}
          pathPrefix: /envoy/verify-params-header
          serverUri:
            cluster: >-
              outbound|80||eas-external-auth-server.external-auth-server.svc.cluster.local
            timeout: 10s
            uri: >-
              http://eas-external-auth-server.external-auth-server.svc.cluster.local
        statusOnError:
          code: Forbidden
        withRequestBody:
          allowPartialMessage: true
          maxRequestBytes: 4096
      filterName: envoy.ext_authz
      filterType: HTTP
      insertPosition:
        index: FIRST
      listenerMatch:
        listenerProtocol: HTTP
        listenerType: SIDECAR_INBOUND
        portNumber: 9097
  workloadLabels:
    app: tekton-dashboard
```

Very neat.. thanks for your work in EAS

**Comment by travisghansen** [comment]
Wow awesome! Can you put this as an example in the examples directory in the project?

**Comment by cameronbraid** [comment]
Sure see https://github.com/travisghansen/external-auth-server/pull/77

**Comment by travisghansen** [comment]
Nice! The multi-arch images take a bit to build but I'll probably have a new tag released today.

**Comment by travisghansen** [comment]
Thanks for the contribution. Merged to master and snapped in `v0.9.0`.
